server/opts.go

```diff
@@ -193,11 +193,31 @@ func parseCluster(cm map[string]interface{}, opts *Options) error {
 			if opts.ClusterTLSConfig, err = GenTLSConfig(tc); err != nil {
 				return err
 			}
-			// For clusters, we will force strict verification. We also act
+
+			// For clusters, we will force strict verification unless the user
+			// specifically disables this by setting `verify: false`. We also act
 			// as both client and server, so will mirror the rootCA to the
 			// clientCA pool.
-			opts.ClusterTLSConfig.ClientAuth = tls.RequireAndVerifyClientCert
-			opts.ClusterTLSConfig.ClientCAs = opts.ClusterTLSConfig.RootCAs
+			//
+			// Because an unspecified verify defaults to false by default we must
+			// first explicitly set the default to true and then fill in the
+			// user-supplied value if it is set in the config.
+			tc.Verify = true
```

Comment:
Instead of having parsing here, which is already done in `parseTLS`, would it make more sense to modify `parseTLS` to accept a boolean which would be the default value for Verify. In `ProcessConfigFile` you would pass false, and here, in `parseCluster` you would pass true. `parseTLS` would set `tc.Verify` to this value before starting the parsing.